src/activity_actor.cpp

```diff
@@ -2139,13 +2140,77 @@ std::unique_ptr<activity_actor> stash_activity_actor::deserialize( JsonIn &jsin
     return actor.clone();
 }
 
+void bookbinder_copy_activity_actor::start( player_activity &act, Character & )
+{
+    act.moves_total = to_moves<int>( pages * 2_minutes );
+    act.moves_left = to_moves<int>( pages * 2_minutes );
+}
+
+void bookbinder_copy_activity_actor::finish( player_activity &act, Character &p )
+{
+    item *it_bkbinder = book_binder.get_item();
```

Comment:
It's possible that this item location can be invalidated, and you end up with a nullptr here.
```suggestion
    item *it_bkbinder = book_binder.get_item();
    if( it_bkbinder == nullptr ) {
        debugmsg( "Invalid bookbinder for copying recipes" );
        act.set_to_null();
        return;
    }
```